### DISCUSSION TITLE
[Help]: Lock-in Slide after drag free swiping

### DISCUSSION THREAD

**Discussion by VMLuca** [davidjerleke/embla-carousel#963] [Help] [answered]
### Summary

Hello,

I want to "lock-in" a slide after swiping with drag free option enabled in order to display the current slide and not to "hang" between two slides.

I have this code which works pretty well besides a delay before calling the event. I know this is expected behaviour as the documentation says: "Please note that this can take longer than you think when dragFree is enabled"

But what can I do to achieve the behavior I descriped without a delay? Is there another approach for dragFree + lock-in?

```js
  embla?.on("settle", () => {
    embla.scrollTo(embla.selectedScrollSnap(), true);
  });`
  ```

### If applicable, which variants of Embla Carousel are relevant to this question?

- [ ] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment]
@VMLuca I don’t have access to my computer for a day or two, but this could possibly work:

```tsx 
emblaApi.on('pointerUp', (emblaApi) => {
  const { scrollTo, target, location } = emblaApi.internalEngine()
  const diffToTarget = target.get() - location.get()

  scrollTo.distance(diffToTarget, true)
})
```

**Reply by VMLuca** [reply]
Great, thanks, I will give it a shot!
But shouldn't there be a option for this behaviour (drag-free + lock-in)?. I can't tell an exact example but I think I stumpled across this behaviour on my phone multiple times on different applications. If so, I would like to open a feature reqquest.

**Reply by davidjerleke** [reply, marked as answer]
@VMLuca there is an option called [skipSnaps](https://www.embla-carousel.com/api/options/#skipsnaps). If you set that to `true` the carousel will skip snaps if dragged in a vigorous manner.

**Reply by VMLuca** [reply]
Ok that's dope! exactly what I looked for! Thanks 👍
